src/vectors.c

```diff
@@ -372,7 +372,7 @@ int zsl_vec_zte(struct zsl_vec *v)
 
 	for (size_t g = 0; g < v->sz; g++) {
 		if ((v->data[g - x] >= 0.0 && v->data[g - x] < epsilon) ||
-		    (v->data[g - x] <= 0.0 && v->data[g - x] > epsilon)) {
+			(v->data[g - x] <= 0.0 && v->data[g - x] > -epsilon)) {
```

Comment:
nitpick: It seems this is a saturation operation, if so I would not use less/greater and equal, since we only want to wrap around to a desired value when the variable surpasses it.